eve/main.yml

```diff
@@ -285,6 +285,134 @@ stages:
+      - SetPropertyFromCommand:
+          name: Set previous version
+          property: metalk8s_prev_version
+          command: >
+            bash -c '
+              source VERSION &&
+              VERSION_MINOR=$((VERSION_MINOR-1))
+            echo $VERSION_MAJOR.$VERSION_MINOR.$VERSION_PATCH$VERSION_SUFFIX'
+      - ShellCommand:
+          name: Retrieve ISO version N-1 image
+          command: >
+            isourl="/pre/metalk8s-%(prop:metalk8s_prev_version)s.iso";
+            curl -s -XGET -o metalk8s.iso
+            "%(prop:artifacts_private_url)s$isourl"
+          haltOnFailure: true
+      - ShellCommand:
+          name: Retrieve ISO N-1 image checksum
+          command: >
+            shasumurl="/pre/SHA256SUM-%(prop:metalk8s_prev_version)s";
+            curl -s -XGET -o SHA256SUM
+            "%(prop:artifacts_private_url)s$shasumurl"
+          haltOnFailure: true
+      - ShellCommand:
+          name: Check N-1 image with checksum
+          command: sha256sum -c SHA256SUM
+          haltOnFailure: true
+      - ShellCommand:
+          name: Create mountpoint
+          command: >
+            sudo mkdir -p /srv/scality/metalk8s-%(prop:metalk8s_prev_version)s
+      - ShellCommand:
+          name: Mount ISO image
+          command: >
+            sudo mount -o loop metalk8s.iso
+            /srv/scality/metalk8s-%(prop:metalk8s_prev_version)s
+          haltOnFailure: true
+      - ShellCommand:
+          name: Create bootstrap configuration file
+          command: |
+            sudo bash << EOF
+            mkdir -p /etc/metalk8s
+            cat > /etc/metalk8s/bootstrap.yaml << END
+            apiVersion: metalk8s.scality.com/v1alpha2
+            kind: BootstrapConfiguration
+            networks:
+              controlPlane: 10.100.0.0/16
+              workloadPlane: 10.100.0.0/16
+            ca:
+              minion: $(hostname)
+            apiServer:
+              host: $(ip route get 10.100.0.0 | awk '/10.100.0.0/{ print $6 }')
+            archives:
+              - /srv/scality/metalk8s-%(prop:metalk8s_prev_version)s
+            END
+            EOF
+      - ShellCommand:
+          name: Start the bootstrap process
+          command: >
+            sudo bash
+            /srv/scality/metalk8s-%(prop:metalk8s_prev_version)s/bootstrap.sh
+            --verbose || true
+          haltOnFailure: true
+      - ShellCommand:
+          name: Retrieve ISO image
+          command: >
+            curl -s -XGET -o /tmp/metalk8s.iso
+            "%(prop:artifacts_private_url)s/metalk8s.iso"
+          haltOnFailure: true
+      - ShellCommand:
+          name: Retrieve ISO image checksum
+          command: >
+            curl -s -XGET -o /tmp/SHA256SUM
+            "%(prop:artifacts_private_url)s/SHA256SUM"
+          haltOnFailure: true
+      - ShellCommand:
+          name: Check image with checksum
+          command: cd /tmp && sha256sum -c SHA256SUM
+          haltOnFailure: true
```

Comment:
IMO you should retrieve and check both ISO at the beginning and use yaml tags see https://github.com/scality/metalk8s/pull/1730/files#r326499412